packages/terminal/src/browser/terminal-widget.ts

```diff
@@ -236,14 +242,16 @@ export class TerminalWidget extends BaseWidget implements StatefulWidget {
      * new terminal widget.
      * If id is provided attach to the terminal for this id.
      */
-    async start(id?: number): Promise<void> {
+    async start(id?: number): Promise<number> {
         await this.waitForResized.promise;
         this.terminalId = typeof id !== 'number' ? await this.createTerminal() : await this.attachTerminal(id);
         if (typeof this.terminalId === "number") {
             await this.doResize();
             this.connectTerminalProcess();
         }
+        return this.terminalId || -1;
```

Comment:
Can you update the function documentation to reflect that it now returns a value?

* Does -1 indicate a failure?
* Wouldn't it be better to return a rejected promise instead?
* Is `0` a valid terminalId?  If so, I guess the expression `this.terminalId || -1` will return -1, which is not what you want.

Reply:
Used second: promise reject.